lightning/src/ln/channelmanager.rs

```diff
@@ -5828,10 +5834,50 @@ where
 					None => continue 'outer_loop,
 				}
 
+				let mut inter_trampoline_path = None;
+
 				// Now process the HTLC on the outgoing channel if it's a forward.
 				if let Some(next_packet_details) = next_packet_details_opt.as_ref() {
+					let first_hop_scid_override = match (&next_packet_details.outgoing_connector, &next_hop) {
+						(HopConnector::Trampoline(next_node_id), onion_utils::Hop::TrampolineForward { ref outer_hop_data, .. } | onion_utils::Hop::TrampolineBlindedForward { ref outer_hop_data, .. }) => {
+							let incoming_amount = outer_hop_data.amt_to_forward;
+							let incoming_cltv = outer_hop_data.outgoing_cltv_value;
+							let outgoing_amount = next_packet_details.outgoing_amt_msat;
+							let outgoing_cltv = next_packet_details.outgoing_cltv_value;
+
+							let routes = self.router.find_route(
+								&self.node_signer.get_node_id(Recipient::Node).unwrap(),
+								&RouteParameters {
+									payment_params: PaymentParameters {
+										payee: Payee::Clear {
+											node_id: next_node_id.clone(),
+											route_hints: vec![],
+											features: None,
+											final_cltv_expiry_delta: 0,
+										},
+										expiry_time: None,
+										max_total_cltv_expiry_delta: incoming_cltv - outgoing_cltv,
+										max_path_count: 1,
+										max_path_length: MAX_PATH_LENGTH_ESTIMATE.div_ceil(2),
+										max_channel_saturation_power_of_half: 2,
+										previously_failed_channels: vec![],
+										previously_failed_blinded_path_idxs: vec![],
+									},
+									final_value_msat: outgoing_amount,
+									max_total_routing_fee_msat: Some(incoming_amount - outgoing_amount),
```

Comment:
Shouldn't the trampoline node itself earn something too?

Reply:
I think it should just include the fee on the local outgoing channel in the total fee during pathfinding? Normally that is not necessary, but this is a payment that is really a forward.